graphistry/pygraphistry.py

```diff
@@ -425,21 +639,40 @@ def register(
         PyGraphistry.client_protocol_hostname(client_protocol_hostname)
         PyGraphistry.certificate_validation(certificate_validation)
         PyGraphistry.store_token_creds_in_memory(store_token_creds_in_memory)
+        PyGraphistry.set_bolt_driver(bolt)
+
         if not (username is None) and not (password is None):
             PyGraphistry.login(username, password, org_name)
-        PyGraphistry.api_token(token or PyGraphistry._config['api_token'])
-        PyGraphistry.authenticate()
+            PyGraphistry.api_token(token or PyGraphistry._config['api_token'])
+            PyGraphistry.authenticate()
+        elif (username is None and not (password is None)):
+            raise Exception(MSG_REGISTER_MISSING_USERNAME)
+        elif not (username is None) and password is None:
+            raise Exception(MSG_REGISTER_MISSING_PASSWORD)
+        elif not (personal_key_id is None) and not (personal_key_secret is None):
+            PyGraphistry.pkey_login(personal_key_id, personal_key_secret)
+            PyGraphistry.api_token(token or PyGraphistry._config['api_token'])
+            PyGraphistry.authenticate()
+        elif personal_key_id is None and not (personal_key_secret is None):
+            raise Exception(MSG_REGISTER_MISSING_PKEY_ID)
+        elif not (personal_key_id is None) and personal_key_secret is None:
+            raise Exception(MSG_REGISTER_MISSING_PKEY_SECRET)
+        elif not (token is None):
+            PyGraphistry.api_token(token or PyGraphistry._config['api_token'])
+        elif not (org_name is None) or is_sso_login:
+            print(MSG_REGISTER_ENTER_SSO_LOGIN)
```

Comment:
stray printf

Reply:
we're fine -- this says if at the end, someone does just `register(org_name='x')` it can assume sso